### PR TITLE
Org archive

### PR DESCRIPTION
**Pull request by CodeMonk**
Couldn't figure out how to pull out an extra commit from master, so had to issue new pull request from branch.

Both your suggestions worked great.

### CODE REVIEW

File: PlainTasks.py [hunk 1]
```diff
@@ -833,3 +837,101 @@ def run(self, edit):
...
+        # only erase our region if the write was successful
+        if success:
+            self.view.erase(edit,region)
+
+        return
+
+    def __writeArchive(self, filename, region):
+        # Write out the given region
+
+        sublime.status_message('Archiving tree to {}'.format(filename))
```
Comment (vovkkk) [inline PlainTasks.py]: `sublime.status_message(u'Archiving tree to {0}'.format(filename))` to support ST2, and all other strings should start with `u` and contain index for format.

File: PlainTasks.py [hunk 2]
```diff
@@ -38,6 +38,10 @@ def run(self, edit):
             self.sys_enc = locale.getpreferredencoding()
         self.project_postfix = self.view.settings().get('project_tag', True)
         self.archive_name = self.view.settings().get('archive_name', 'Archive:')
+        # org-mode style archive stuff
+        self.archive_org_default_filemask = "{dir}{sep}{base}_archive{ext}"
```
Comment (vovkkk) [inline PlainTasks.py]: `u` too

File: PlainTasks.py [hunk 3]
```diff
@@ -833,3 +837,101 @@ def run(self, edit):
...
+            archive_filename=self.archive_org_default_filemask.format(
+                    dir=dir, base=base, ext=extension, sep=sep)
+
+            # Display error, letting the user know
+            sublime.error_message("Error:\n\nInvalid filemask:{}\nUsing default: {}".format(
+                self.archive_org_filemask, self.archive_org_default_filemask))
+
+        return archive_filename
+
+    def __regionIndentLen(self, region):
```
Comment (vovkkk) [inline PlainTasks.py]: unused method

File: PlainTasks.py [hunk 4]
```diff
@@ -833,3 +837,101 @@ def run(self, edit):
...
+    def __findCurrentSubtree(self):
+        # Return the region that starts at the cursor, or starts at
+        # the beginning of the selection
+
+        line = self.view.line(self.view.sel()[0].begin())
+        # Start finding the region at the beginning of the next line
+        region = self.view.indented_region(line.b+2)
+
+        # It missed our current line, so, back up one line.
+        region.a=line.a
```
Comment (vovkkk) [inline PlainTasks.py]:
```python
        if not region.empty():
            region = sublime.Region(line.a, region.b)
```

**Comment by vovkkk** [comment]
You can commit and push in the same branch

**Comment by CodeMonk** [comment]
All changes made.  Pushing.

I didn't branch before my work, and then I started working on my next little thing.  I couldn't figure out how to remove a commit from master, for the pull request, and put that into a branch.  So, instead, I branched at my commit for org-archive, and re-submitted the pull.

Sorry -- I'm normally a mercurial guy, and git has its nuances.

**Comment by vovkkk** [comment]
You forgot indexes `{0}{1}` instead of `{}{}` in formated strings, but alright I’ll merge by hand since need to update readme and tutorial too.

Thank you for pull request.

**Comment by CodeMonk** [comment]
Couple of questions:

I'm going to be adding a couple more changes.  Should I file issue requests first to discuss them?

As to some of the compatibility stuff:  unicode {0} vs {}, {} vs "%s" % str -- is there a style guide somewhere?  Or, maybe just use pylint?

**Comment by vovkkk** [comment]
> unicode {0} vs {}, {} vs "%s" % str -- is there a style guide somewhere? Or, maybe just use pylint?

I use ST2 as main editor, the best way to know, afaik, if code compatible is to run it in both versions; others incompatibilities out of head: `filter()` returns list in py2 and generator in py3, dictionary comprehension was added in py2.7.sth while ST2 has py2.6.sth so you have to use `dict()`, import module from same directory in ST3 has to start with dot `import .module` while in ST2 just `import module`, some ST API methods differs e.g. ST3 has `view.is_valid()` ST2 has no.

**Comment by CodeMonk** [comment]
Oh -- you actually had issues with ST2 -- I'll be sure and test changes on
both!

I just assumed you knew that {} gives issues on python 2.6

**Pull request closed without merging**